clients/go/gengo.go

```diff
@@ -72,38 +85,196 @@ type WagClient struct {
 	// Keep the circuit doer around so that we can turn it on / off
 	circuitDoer    *circuitBreakerDoer
 	defaultTimeout time.Duration
-	logger       logger.KayveeLogger
+	logger      waglogger.WagClientLogger
 }
 
 var _ Client = (*WagClient)(nil)
 
+
+//This pattern is used instead of using closures for greater transparency and the ability to implement additional interfaces.
+type options struct {
+	transport    http.RoundTripper
+	logger       waglogger.WagClientLogger
+	instrumentor Instrumentor
+	exporter     sdktrace.SpanExporter
+}
+
+type Option interface {
+	apply(*options)
+}
+
+//Logger
+
+//WithLogger sets client logger option.
+func WithLogger(log waglogger.WagClientLogger) Option {
+	return loggerOption{Log: log}
+}
+
+type loggerOption struct {
+	Log waglogger.WagClientLogger
+}
+
+func (l loggerOption) apply(opts *options) {
+	opts.logger = l.Log
+}
+
+//RoundTripper
+
+type roundTripperOption struct {
+	rt http.RoundTripper
+}
+
+func (t roundTripperOption) apply(opts *options) {
+	opts.transport = t.rt
+}
+
+//WithRoundTripper allows you to pass in intrumented/custom roundtrippers which will then wrap the
```

Comment:
nit but most of our Go code comments have a space between the `//` and the start of the comment, i.e.
```suggestion
// WithRoundTripper allows you to pass in intrumented/custom roundtrippers which will then wrap the
```